src/cow-react/modules/swap/services/safeBundleFlow/index.ts

```diff
@@ -0,0 +1,81 @@
+import { logTradeFlow } from '@cow/modules/trade/utils/logger'
+import { SafeBundleFlowContext } from '@cow/modules/swap/services/types'
+import { PriceImpact } from '@src/hooks/usePriceImpact'
+import { Percent } from '@uniswap/sdk-core'
+import { buildApproveTx } from '@cow/modules/operations/bundle/approve'
+import { buildPresignTx } from '@cow/modules/operations/bundle/presign'
+import { signAndPostOrder } from '@src/utils/trade'
+import { MetaTransactionData } from '@safe-global/safe-core-sdk-types'
+import { getSwapErrorMessage } from '@cow/modules/trade/utils/swapErrorHelper'
+// import { tradeFlowAnalytics } from '@cow/modules/trade/utils/analytics'
+
+const LOG_PREFIX = 'SAFE BUNDLE FLOW'
+
+export async function safeBundleFlow(
+  input: SafeBundleFlowContext,
+  priceImpactParams: PriceImpact,
+  confirmPriceImpactWithoutFee: (priceImpact: Percent) => Promise<boolean>
+): Promise<void> {
+  logTradeFlow(LOG_PREFIX, 'STEP 1: confirm price impact')
+
+  if (priceImpactParams?.priceImpact && !(await confirmPriceImpactWithoutFee(priceImpactParams.priceImpact))) {
+    return
+  }
+
+  try {
+    // For now, bundling ALWAYS includes these 2 steps.
+    // In the feature users will be able to sort/add steps as they see fit
+    logTradeFlow(LOG_PREFIX, 'STEP 2: approval tx')
+    const approveTx = await buildApproveTx({
+      erc20Contract: input.erc20Contract,
+      spender: input.spender,
+      amountToApprove: input.context.trade.inputAmount,
+    })
+    logTradeFlow(LOG_PREFIX, 'STEP 3: post order')
+    const { id: orderId, order: _order } = await signAndPostOrder(input.orderParams).finally(() => {
+      input.callbacks.closeModals()
+    })
+
+    logTradeFlow(LOG_PREFIX, 'STEP 4: ORDER', orderId, _order)
+    logTradeFlow(LOG_PREFIX, 'STEP 4: presign tx')
+    const presignTx = await buildPresignTx({ settlementContract: input.settlementContract, orderId })
+    logTradeFlow(LOG_PREFIX, 'STEP 5: send safe tx')
+    const safeTransactionData: MetaTransactionData[] = [
+      { to: approveTx.to!, data: approveTx.data!, value: '0', operation: 0 },
+      { to: presignTx.to!, data: presignTx.data!, value: '0', operation: 0 },
+    ]
+    // Get safe sdk instance
+    const safeSdk = input.safeSdk
+    const safeTx = await safeSdk.createTransaction({ safeTransactionData, onlyCalls: true })
+    logTradeFlow(LOG_PREFIX, 'STEP 5: built safe tx', safeTx)
+
+    const safeTxHash = await safeSdk.getTransactionHash(safeTx)
+    logTradeFlow(LOG_PREFIX, 'STEP 5: safe tx hash', safeTxHash)
+
+    const txhash = await input.provider.send('eth_sendTransaction', [
+      { to: safeTx.data.to, data: safeTx.data.data, value: safeTx.data.value },
+    ])
```

Comment:
This is the "meat" of the PR and where likely the issue is.

Building the approval and pre-sign txs, creating the safe tx, sending it with `eth_sendTransaction`